062-support-tiered-storage.md

```diff
@@ -0,0 +1,121 @@
+#  Support tiered storage natively in Strimzi
+
+## Current situation
+
+Kafka introduced tiered storage feature since 3.6. The feature is in early stage and it’s nice to support tiered storage configuration natively via Strimzi.
+
+## Motivation
+
+Supporting tiered storage via Strimzi allow Strimzi users to enable tiered storage natively with Strimzi config. By allowing to config custom configuration for RemoteStorageManager and RemoteLogMetadataManager, users are able to adopt tiered storage feature on Strimzi easily.
+
+It should also be noted that, it’s on the end-user to ensure that their tiered storage setup works with Strimzi; the operator itself is strictly responsible for pushing down the necessary config to the broker pods.
+
+## Proposal
+
+### RemoteStorage
+
+Add a RemoteStorage class under storage config (under https://strimzi.io/docs/operators/latest/configuring.html#type-KafkaClusterSpec-schema-reference), which would support configuration for all tiered storage related configuration.
+
+The RemoteStorage class contains configuration for various aspects, including system config, RSM specific config, RLMM specific config.
+
+```
+kafka:
+  storage:
+    remote:
+      enabled: true
+      rlm:
+        # See below detail
+      rsm:
+        # See below detail
+      rlmm:
+        # See below detail
```

Comment:
I wonder if there should be some new `tieredStorage` option rather then pushing this into the `storage` section. Especially given the the current `storage` will be gone from the Kafka CR soon and move into the `KafkaNodePool` resources, but the tired storage configuration should probably stay inside the `Kafka` CR.

You might also need to use some `type: custom` differentiator or have some `customTieredStorage` subsection (e.g. `.spec.kafka.tiredStorage.customTieredStorage`). The presence or absence of this section would define if it i enbaled or disabled. So I do not think you need the `enabled` field.

Reply:
Done